lit_llama/utils.py

```diff
@@ -314,3 +314,160 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         del self.zf  # I don't think there is a way to force closing...
         self.zf = None
+
+
+class SavingProxyForStorage:
+    def __init__(self, obj, saver, protocol_version=5):
+        self.protocol_version = protocol_version
+        self.saver = saver
+        if not (isinstance(obj, torch.storage.TypedStorage) or torch.is_storage(obj)):
+            raise TypeError(f"expected storage, not {type(obj)}")
+
+        # this logic is taken from PyTorch 2.0+ torch/serialization.py
+        if isinstance(obj, torch.storage.TypedStorage):
+            # PT upstream wants to deprecate this eventually...
```

Comment:
I see that TypedStorage is already deprecated in `torch/storage.py`. Can you make sure that we aren't raising internal deprecation errors to users of lit-llama?

Reply:
I think we are not seeing any because we are not using `Tensor.storage()` (or otherwise creating TypedStorages).
Do you see any deprecation messages (in the test or conversion)?

Reply:
:sweat_smile: